### PR TITLE
NT-1916: Feature flag comment threading

### PR DESCRIPTION
**Pull request by Arkariang**
# 📲 What

- Added feature Flag on Optimizely
- Added new attributes to optimizaly calls
 `"session_app_release_version_number" = 2131` version (2.13.1) in numbers,
 `"session_app_release_version_code" = BuildConfig.VERSION_CODE`

# 🤔 Why
- Feature Flag for the new Treaded Coment Screen
- Adding the new attributes will provide more flexibility around versioning for feature flags

# 👀 See
Screen Shot 2021-05-12 at 2 51 36 PM
- Feature Flag On
FeatureFlagOn
- Feature Flag Off
FeatureFlagOff

|  |  |

# 📋 QA
- Enable the feature flag, when you press the comments row in the project page no new screen will be presented, that screen is work in progress
- Disable the feature flag, the old comments screen will be presented 

# Story 📖

[NT-1916](https://kickstarter.atlassian.net/browse/NT-1916)

### CODE REVIEW

File: app/src/main/java/com/kickstarter/ui/activities/ProjectActivity.kt [hunk 1]
```diff
@@ -665,6 +670,10 @@ class ProjectActivity :
         startActivityWithTransition(intent, R.anim.slide_in_right, R.anim.fade_out_slide_out_left)
     }
 
+    private fun startCommentsThreadedActivity(projectAndData: Pair<Project, ProjectData>) {
+        // TODO: Start the new activity defined in https://kickstarter.atlassian.net/browse/NT-1920
```
Comment (Arkariang) [inline app/src/main/java/com/kickstarter/ui/activities/ProjectActivity.kt:674]: @leighdouglas you can start the new activity in this method, for the task https://kickstarter.atlassian.net/browse/NT-1920

**Review by sunday-okpoluaefe — APPROVED** [review]
lgtm!

File: app/src/main/java/com/kickstarter/ui/activities/ProjectActivity.kt (hunk 1 above)
Comment (leighdouglas) [reply app/src/main/java/com/kickstarter/ui/activities/ProjectActivity.kt:674]: Got it! will do as soon as this lands and will update my PR

File: app/src/main/java/com/kickstarter/viewmodels/ProjectViewModel.kt [hunk 2]
```diff
@@ -1069,6 +1082,9 @@ interface ProjectViewModel {
         @NonNull
         override fun startCommentsActivity(): Observable<Pair<Project, ProjectData>> = this.startCommentsActivity
 
+        @NonNull
+        override fun startCommentsThreadedActivity(): Observable<Pair<Project, ProjectData>> = this.startCommentsThreadedActivity
```
Comment (leighdouglas) [inline app/src/main/java/com/kickstarter/viewmodels/ProjectViewModel.kt:1086]: Nit: I think to keep parity with iOS, we are eventually going to name the comments activity `RootCommentsActivity`, so you may want to change the name of this method from threaded to root

Comment (Arkariang) [reply app/src/main/java/com/kickstarter/viewmodels/ProjectViewModel.kt]: Done!

**Pull request merged**